nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -1048,6 +1048,30 @@ contract StakingEscrow is Issuer, IERC900History {
         }
     }
 
+    /**
+    * @notice Remove unused sub-stake to decrease gas cost for several methods
+    */
+    function removeUnusedSubStake(uint16 _index) external onlyStaker {
+        StakerInfo storage info = stakerInfo[msg.sender];
+
+        uint256 lastIndex = info.subStakes.length - 1;
+        SubStakeInfo storage subStake = info.subStakes[_index];
+        require(subStake.lastPeriod != 0 &&
+                (info.currentCommittedPeriod == 0 ||
+                subStake.lastPeriod < info.currentCommittedPeriod) &&
+                (info.nextCommittedPeriod == 0 ||
+                subStake.lastPeriod < info.nextCommittedPeriod));
+
+        if (_index != lastIndex) {
+            SubStakeInfo storage lastSubStake = info.subStakes[lastIndex];
+            subStake.firstPeriod = lastSubStake.firstPeriod;
+            subStake.lastPeriod = lastSubStake.lastPeriod;
+            subStake.periods = lastSubStake.periods;
+            subStake.lockedValue = lastSubStake.lockedValue;
+        }
+        info.subStakes.pop();
```

Comment:
Does using `pop` mean that the function can only remove one unused sub-stake at a time? Unless I'm mistaken this just deletes one element from the end of an array.

If so, how does one keep track of their removal of multiple sub-stakes? It doesn't show you unused sub-stakes when you check via `nucypher stake list`

Reply:
You're right - this deletes last element in array but before this - copies last element to place of removed unused sub-stake.
To show inactive sub-stakes you can use option `--all` for `nucypher stake list` or interactive mode for `nucypher stake remove-unused`